### ISSUE TITLE
Can’t build with unclear cause ‘Auth fail’

### ISSUE THREAD

**Issue by vovtz** [shillner/unleash-maven-plugin#62]
My colleague is having the same issue I had a few weeks ago when trying to build with ‘unleash’ from a Jenkins job. I remember having wasted too much time and ending up with a working configuration without ever finding out for sure what caused the problem in the first place.

His problem appears to be equally stubborn and he ended up looking at the source code for the ‘unleash-scm-provider-git’ plug-in. From the stack trace (see below) we are unable to determine the exact cause. The original exception thrown is unfortunately neither logged nor re-thrown. His public key is correctly configured in GitLab (he can pull from the command line) and the host key for the GitLab server is in his ‘know_hosts’ file (again, because he can successfully pull and push from the CLI).

We don’t know what file the plug-in uses for the private key or what username is used for the SSH session. It looks like the plug-in isn’t using the system’s SSH implementation and configuration, because the INFO logging level we configured is ignored. The message ‘Auth fail’ is fairly unspecific and useless for debugging purposes :(

```
…
[INFO] 16:16:20,032  Git - Searching for Git tag 'rni-parent-3.31.0'
[DEBUG] 16:16:20,041  Git - Query info:
  - TAG_NAME: rni-parent-3.31.0
  - WORKING_DIR: /Users/jaap/dev/workspace/rni/rni
  - REMOTE: origin
  - REMOTE_URL: git@www.gitlab.rni.ictu:main/rni.git
[ERROR] An exception was caught while processing the workflow step with id 'tagScm'.
com.itemis.maven.plugins.unleash.scm.ScmException: An error occurred while querying the remote git repository for tag 'rni-parent-3.31.0'.
  at com.itemis.maven.plugins.unleash.scm.providers.ScmProviderGit.hasTag(ScmProviderGit.java:623)
  at com.itemis.maven.plugins.unleash.steps.actions.TagScm.execute(TagScm.java:88)
  at com.itemis.maven.plugins.cdi.internal.util.workflow.WorkflowExecutor.executeSequentialWorkflowStep(WorkflowExecutor.java:157)
  at com.itemis.maven.plugins.cdi.internal.util.workflow.WorkflowExecutor.execute(WorkflowExecutor.java:119)
  at com.itemis.maven.plugins.cdi.AbstractCDIMojo.execute(AbstractCDIMojo.java:209)
  at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
  at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
  at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
  at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
  at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
  at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
  at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
  at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
  at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
  at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
  at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
  at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
  at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
  at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
  at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:498)
  at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
  at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
  at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
  at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.eclipse.jgit.api.errors.TransportException: git@www.gitlab.rni.ictu:main/rni.git: Auth fail
  at org.eclipse.jgit.api.LsRemoteCommand.execute(LsRemoteCommand.java:220)
  at org.eclipse.jgit.api.LsRemoteCommand.call(LsRemoteCommand.java:159)
  at com.itemis.maven.plugins.unleash.scm.providers.ScmProviderGit.hasTag(ScmProviderGit.java:617)
  ... 26 more
Caused by: org.eclipse.jgit.errors.TransportException: git@www.gitlab.rni.ictu:main/rni.git: Auth fail
  at org.eclipse.jgit.transport.JschConfigSessionFactory.getSession(JschConfigSessionFactory.java:159)
  at org.eclipse.jgit.transport.SshTransport.getSession(SshTransport.java:136)
  at org.eclipse.jgit.transport.TransportGitSsh$SshFetchConnection.<init>(TransportGitSsh.java:264)
  at org.eclipse.jgit.transport.TransportGitSsh.openFetch(TransportGitSsh.java:162)
  at org.eclipse.jgit.api.LsRemoteCommand.execute(LsRemoteCommand.java:198)
  ... 28 more
Caused by: com.jcraft.jsch.JSchException: Auth fail
  at com.jcraft.jsch.Session.connect(Session.java:491)
  at org.eclipse.jgit.transport.JschConfigSessionFactory.getSession(JschConfigSessionFactory.java:116)
  ... 32 more
[INFO] Rolling back after execution errors - please find the error messages and stack traces above.
[INFO] 16:16:20,676  Rollback of SCM tag creation and POM modifications due to a processing exception.
[INFO] 16:16:20,676  Git - Searching for Git tag 'rni-parent-3.31.0'
[DEBUG] 16:16:20,677  Git - Query info:
…
```

I hope you can shed some light on this situation. And perhaps you can give us a hint how to change the logging level.

**Comment by shillner** [comment]
Hi, currently it is hardly possible for me to spend some time on this but I know it worked well some time ago. I'll try to get into this error soon and I know that the git provider code looks fairly ugly :(

1. Have you tried the `scmSshPassphrase` parameter?
2. Have you tried running the whole thing with an SSH agent?
3. I'll likely try the whole thing out with our internal gitlab but just in case it is not reproducible, could you provide a simple test configuration for me so that I'm able to debug this issue?

**Comment by prinsj** [comment]
The problem has been solved. I found an issue in the jgit forum that actually had the solution: https://www.eclipse.org/forums/index.php/m/1689963/?srch=auth+fail#msg_1689963.
It turns out that changing the ~/.ssh/config file to this:
```
Host *
	UserKnownHostsFile	~/.ssh/known_hosts
	IdentityFile ~/.ssh/id_rsa
	HashKnownHosts no
```
did the trick.

**Comment by shillner** [comment]
I had some time to investigate this behavior and found another bug in the git provider (see linked issue). This one is now fixed in version 2.0.3 of the git provider.

Thanks for posting the solution. Closing this issue.
